ocaml/xapi/xapi.ml

```diff
@@ -559,6 +564,33 @@ let resynchronise_ha_state () =
     error "Caught exception resynchronising state of HA system: %s"
       (ExnHelper.string_of_exn e)
 
+let maybe_set_fqdn_in_pool_conf () =
+  let open Pool_role in
+  match get_role () with
+  | Slave current_master_address ->
+      Server_helpers.exec_with_new_task "Set fqdn in pool.conf"
+        (fun __context ->
+          let master_hostname =
+            Helpers.get_master ~__context |> fun self ->
+            Db.Host.get_hostname ~__context ~self
+          in
+          match Gencertlib.Lib.fqdn_of_hostname master_hostname with
+          | Some master_fqdn ->
+              if current_master_address <> master_fqdn then (
+                D.info
+                  "set_fqdn_in_pool_conf: updating pool.conf with master_fqdn: \
+                   %s"
+                  master_fqdn ;
+                Slave master_fqdn |> Xapi_pool_transition.set_role
+              ) else
+                D.info "set_fqdn_in_pool_conf: not modifying pool.conf"
+          | None ->
+              D.error
```

Comment:
This logs an error but does not communicate the error to the caller. How does the caller become aware of the problem?

Reply:
This is being called as part of the xapi startup sequence, so if it raised then I believe it would take down xapi's main thread. We could run this in a separate thread, but that seems like overkill